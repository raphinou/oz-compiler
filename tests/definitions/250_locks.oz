% Locks
local
   L
   C
in
   L={NewLock}
   C={NewCell 0}
   thread
      {Delay 500}
      lock L then
	 {Show 't1 got lock'}
	 C:=@C+1
	 {Show @C}
      end
   end
   thread
      {Delay 500}
      lock L then
	 {Show 't2 got lock'}
	 C:=@C+2
	 {Show @C}
      end
   end
   thread
      {Delay 700}
      {Show 't1 and t2 try to get lock, but it is not available'}
   end
   thread
      lock L then
	 {Show 't3 got lock and waits 1s'}
	 {Delay 1000}
	 C:=@C+3
	 {Show @C}
      end
   end
end
