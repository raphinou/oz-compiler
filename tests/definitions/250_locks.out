't3 got lock and waits 1s'
't1 and t2 try to get lock, but it is not available'
3
't2 got lock'
5
't1 got lock'
6
